### PR TITLE
Add database migration with yuniql

### PR DESCRIPTION
**Pull request by SandGrainOne**
## Description
Add database migration to Profile startup with the help of Yuniql.

## Related Issue(s)
- #190

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Profile.Integrations/Extensions/WebApplicationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    /// <param name="config">the configuration collection</param>
+    public static void SetUpPostgreSql(this IApplicationBuilder app, bool isDevelopment, IConfiguration config)
+    {
+        PostgreSqlSettings? settings = config.GetSection("PostgreSQLSettings")
+            .Get<PostgreSqlSettings>()
+            ?? throw new ArgumentNullException(nameof(config), "Required PostgreSQLSettings is missing from application configuration");
+
+        if (settings.EnableDBConnection)
+        {
+            ConsoleTraceService traceService = new() { IsDebugEnabled = true };
```
Comment (hggutvik) [inline src/Altinn.Profile.Integrations/Extensions/WebApplicationExtensions.cs:33]: So this line should run in any environment, and not just in local dev? Just making sure that the "debug enabled" won't cause troubles in prod

File: src/Altinn.Profile.Integrations/Migration/v0.01/02-setup-tables.sql [hunk 2]
```diff
@@ -37,6 +30,3 @@ CREATE TABLE IF NOT EXISTS contact_and_reservation.person (
     CONSTRAINT fk_mailbox_supplier FOREIGN KEY (mailbox_supplier_id_fk) REFERENCES contact_and_reservation.mailbox_supplier (mailbox_supplier_id),
     CONSTRAINT chk_language_code CHECK (language_code ~* '^[a-z]{2}$')
 );
-
--- Indexes for performance
-CREATE INDEX idx_fnumber_ak ON contact_and_reservation.person (fnumber_ak);
```
Comment (hggutvik) [inline src/Altinn.Profile.Integrations/Migration/v0.01/02-setup-tables.sql:42]: Make sure that this index is no longer needed here

**Review by hggutvik — APPROVED** [review]
Approving, with some comments/questions on thing to verify before closing the PR

File: src/Altinn.Profile.Integrations/Extensions/WebApplicationExtensions.cs (hunk 1 above)
Comment (SandGrainOne) [reply src/Altinn.Profile.Integrations/Extensions/WebApplicationExtensions.cs:33]: Debug is disabled by default.

File: src/Altinn.Profile.Integrations/Migration/v0.01/02-setup-tables.sql (hunk 2 above)
Comment (SandGrainOne) [reply src/Altinn.Profile.Integrations/Migration/v0.01/02-setup-tables.sql:42]: Good catch, this was not intentional.

**Pull request merged**